packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -0,0 +1,123 @@
+import * as cdk from '@aws-cdk/core';
+import { TagOption } from '../common';
+import { TagUpdatesOptions } from '../constraints';
+import { IPortfolio, Portfolio } from '../portfolio';
+import { IProduct, Product } from '../product';
+import { CfnPortfolioProductAssociation, CfnResourceUpdateConstraint, CfnTagOption, CfnTagOptionAssociation } from '../servicecatalog.generated';
+import { hashValues } from './util';
+import { InputValidator } from './validation';
+
+export class AssociationManager {
+  public static associateProductWithPortfolio(scope: cdk.Resource, portfolio: IPortfolio, product: IProduct) {
+    const associationKey = hashValues(portfolio.node.addr, product.node.addr, scope.stack.node.addr);
+    if (!this.associationMap.has(associationKey)) {
+      const association = new CfnPortfolioProductAssociation(scope, `PortfolioProductAssociation${associationKey}`, {
+        portfolioId: portfolio.portfolioId,
+        productId: product.productId,
+      });
+      this.associationMap.set(associationKey, new Map());
+      this.associationMap.get(associationKey).set(Constraints.ASSOCIATION, association);
+    }
+  }
+
+  public static addResourceUpdateConstraint(scope: cdk.Resource, product: IProduct, options: TagUpdatesOptions) {
+    const [pair, associationKey] = this.associationPrecheck(scope, product);
+    InputValidator.validateLength(this.generateAssocationString(scope, pair), 'description', 0, 2000, options.description);
+    if (!this.associationMap.get(associationKey).get(Constraints.RESOURCE_UPDATE)) {
+      const constraint = new CfnResourceUpdateConstraint(scope, `ResourceUpdateConstraint${associationKey}`, {
+        acceptLanguage: options.acceptLanguage,
+        description: options.description,
+        portfolioId: pair.portfolio.portfolioId,
+        productId: pair.product.productId,
+        tagUpdateOnProvisionedProduct: options.tagUpdateOnProvisionedProductAllowed || options.tagUpdateOnProvisionedProductAllowed === undefined
+          ? Allowed.ALLOWED : Allowed.NOT_ALLOWED,
```

Comment:
This is a weird way to phrase it... how about

```suggestion
        tagUpdateOnProvisionedProduct: options.tagUpdateOnProvisionedProductAllowed === false 
          ? Allowed.NOT_ALLOWED
          : Allowed.ALLOWED,
```